YubiKit/YubiKit/Connections/NFCConnection/YKFNFCConnection.m

```diff
@@ -293,11 +293,14 @@ - (void)updateServicesForSession:(NFCTagReaderSession *)session tag:(id<NFCISO78
         return;
     }
     
+    YKFNFCConnectionState previousState = self.nfcConnectionState;
     self.nfcConnectionState = state;
 
     switch (state) {
         case YKFNFCConnectionStateClosed:
-            [self.delegate didDisconnectNFC:self error:self.nfcConnectionError];
+            if (previousState == YKFNFCConnectionStateOpen) {
```

Comment:
If this is changed, will there be any feedback to the consuming app to indicate that the connection prompt has been dismissed? 
The consumer may be in a state where it believes the prompt is still being displayed when the prompt is actually gone.

Reply:
The reasoning behind this change was that not getting a connection would be slightly strange to signal with a call too `didDisconnectNFC(_ connection: YKFNFCConnection, error: Error?)`. I've reverted back to previous logic so even if we don't disconnect from anything we still get the error via the callback when a user cancels the NFC modal. I don't like either solution so we might revisit this in the future.

Reply:
Mixing delegate callbacks and completion handlers does seem like a clash of styles.

Reply:
I'm not sure I understand. Where are the completion handlers?

Reply:
We tried both a pure delegate based api, blocks only and the mixed version we have now. They all have their own advantages and disadvantages but the mixed version ended up being the most convenient one when using the SDK in real applications.